tests/relay/server.js

```diff
@@ -0,0 +1,185 @@
+  if (typeof accountSid !== 'string') {
+    throw new Error('ACCOUNT_SID env var not defined.');
+  }
+
+  return { apiKeySid, apiKeySecret, accountSid };
+};
+
+/**
+ * @typedef {(callSid: string, callbackUrl: string) => Promise<import('axios').AxiosResponse>} CreateSubscription
+ */
+
+/**
+ * @typedef {(callSid: string) => Promise<import('axios').AxiosResponse>} SendMessage
+ */
+
+/**
+ * @param {string} accountSid
+ * @param {string} apiKeySid
+ * @param {string} apiKeySecret
+ */
+const createTwilioUserDefinedMessageActions = (accountSid, apiKeySid, apiKeySecret) => {
+  const baseUrl = 'https://api.twilio.com/2010-04-01';
+  const extendedUrl = `${baseUrl}/Accounts/${accountSid}/Calls`;
+
+  /**
+   * @type {CreateSubscription}
+   */
+  const createSubscription = (callSid, callbackUrl) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessageSubscriptions.json`;
+
+    const params = new URLSearchParams();
+    params.append('Method', 'POST');
+    params.append('Callback', callbackUrl);
+
+    return axios.post(url, params, {
+      auth: {
+        username: apiKeySid,
+        password: apiKeySecret,
+      },
+    });
+  };
+
+  /**
+   * @type {SendMessage}
+   */
+  const sendMessage = (callSid) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessages.json`;
+
+    const params = new URLSearchParams();
+    params.append('Content', JSON.stringify({ message: 'ahoy, world!' }));
+
+    return axios.post(url, params, {
+      auth: {
+        username: apiKeySid,
+        password: apiKeySecret,
+      },
+    });
+  };
+
+  return { createSubscription, sendMessage };
+}
+
+/**
+ * @param {CreateSubscription} createSubscription
+ * @param {SendMessage} sendMessage
+ */
+const createExpressApp = (createSubscription, sendMessage) => {
+  const app = express();
+
+  app.use(cors());
+  app.use(express.json());
+  app.use(express.urlencoded({ extended: true }));
+
+  app.post('/create-subscription', async (req, res) => {
+    const { CallSid } = req.body;
+    if (typeof CallSid !== 'string') {
+      res.sendStatus(400);
+      throw new Error('CallSid is not of type string');
+    }
+    const callbackUrl = `${app.listener.url()}/receive-message`;
+    console.log('creating a subscription', { CallSid, callbackUrl });
+
+    const subscription = await createSubscription(CallSid, callbackUrl);
+    console.log(subscription.data);
+
+    res.json(subscription.data);
+  });
+
+  app.post('/receive-message', (req, res) => {
+    console.log('received message', req.body);
+
+    const { CallSid } = req.body;
+    if (typeof CallSid !== 'string') {
+      res.sendStatus(400);
+      throw new Error('CallSid is not of type string');
+    }
+
+    const existingMessages = receivedMessages.get(CallSid) || [];
+    receivedMessages.set(CallSid, [...existingMessages, req.body]);
+
+    res.sendStatus(200);
+  });
+
+  app.post('/get-received-messages', (req, res) => {
```

Comment:
Since we're retrieving, this should be a GET API. The parameter can be passed via query or path.
```
app.get('/received-message')
```

Reply:
If I recall correctly, GET requests do some caching right? I used a POST here so I can avoid having to write out special headers for this route. Correct me if that assumption is wrong.

Reply:
Correct, GET are cached by browsers. The proper way is to set cache headers. It's better to set a cache header for this single endpoint than making it a POST and call it `/get-something` :)
```
res.setHeader("Cache-Control", "no-cache");
```